src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(T item, Node<T> prev, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(value, null, null);
+            tail = head;
+            size++;
+        } else {
+            Node<T> newNode = new Node<>(value, tail, null);
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node selectedNode = getNodeById(index);
+        if (selectedNode == head) {
+            Node newNode = new Node<>(value, null, selectedNode);
+            selectedNode.prev = newNode;
+            head = newNode;
+        } else {
+            Node newNode = new Node<>(value, selectedNode.prev, selectedNode);
+            selectedNode.prev.next = newNode;
+            selectedNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (size / 2 < index) {
+            Node<T> currentNode = tail;
+            for (int i = size - 1; i >= 0; i--) {
+                if (index == i) {
+                    return currentNode.item;
+                }
+                currentNode = currentNode.prev;
+            }
+        } else {
+            Node<T> currentNode = head;
+            for (int i = 0; i < size; i++) {
+                if (index == i) {
+                    return currentNode.item;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        throw new IndexOutOfBoundsException();
```

Comment:
It looks very bulky. You have the same logic in your getNodeById() method, just use this method here (maybe fix a some lines in getNodeById to make this method universal